system/lib/wasmfs/wasmfs.cpp

```diff
@@ -119,4 +128,57 @@ __wasi_errno_t __wasi_fd_read(
   *nread = num;
   return __WASI_ERRNO_INVAL;
 }
+
+__wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
+  int accessMode = (flags & O_ACCMODE);
+  bool canWrite = false;
+
+  if (accessMode == O_WRONLY || accessMode == O_RDWR) {
+    canWrite = true;
+  }
+
+  std::string interim = "";
+  std::vector<std::string> pathParts;
+
+  auto newPathName = (const char*)pathname;
+
+  for (int i = 0; newPathName[i] != '\0'; i++) {
+    if (newPathName[i] != '/') {
+      interim += newPathName[i];
+    } else {
+      // Avoid adding empty string
+      if (i > 0) {
+        pathParts.push_back(interim);
+      }
+      interim = "";
+    }
+  }
+  // Case where there is no / at the end of the string.
+  pathParts.push_back(interim);
+
+  std::shared_ptr<File> curr = getRootDirectory();
+  for (int i = 0; i < pathParts.size(); i++) {
+
+    // If it is a file and we have not reached the end of the path, exit
+    if (curr->is<DataFile>() && i != pathParts.size() - 1) {
+      return -(EBADF);
+    }
+
+    // Find the next entry in the current directory entry
+#ifdef WASMFS_DEBUG
+    curr->get().printKeys();
+#endif
+    curr = curr->cast<Directory>()->get().getEntry(pathParts[i]);
+
+#ifdef WASMFS_DEBUG
+    std::vector<char> temp(pathParts[i].begin(), pathParts[i].end());
+    emscripten_console_log(&temp[0]);
+#endif
```

Comment:
Is `WASMFS_DEBUG` for your own debugging now @ethanalee or is this for logging of called syscalls + return values?

(If it's for the latter, I'd have some suggestions. But that can be left for later.)

Reply:
The former, I think we do want to log syscalls + return values however. I just put this flag because it was cleaner than just putting a `// TODO: remove later`.